src/DependencyInjection/SurfnetSamlExtension.php

```diff
@@ -158,9 +158,14 @@ private function parseRemoteIdentityProviderConfiguration(array $identityProvide
         $configuration = [
             'entityId' => $identityProvider['entity_id'],
             'ssoUrl' => $identityProvider['sso_url'],
-            'certificateData' => $identityProvider['certificate'],
         ];
 
+        if ( isset($identityProvider['certificate_file']) ) {
+            $configuration['certificateFile'] = $identityProvider['certificate_file'];
+        } else {
+            $configuration['certificateData'] = $identityProvider['certificate'];
+        }
```

Comment:
Nice :)

Note that by allowing a certificate *file* to be configured, you will [get run-time exceptions](https://github.com/SURFnet/saml2/blob/a7e5b2fdd849a98ed86312baad2d9ac660bdbdc4/src/SAML2/Utilities/File.php#L24-L37) during authentications when the file does not exist on the deployment target.